[Clifford Kaleb Toh Jing Zhe] iP

Droffilc13 opened the pull request:
# DukePro

As Benjamin Franklin said,
> If you fail to plan, 
> you are planning to fail!

Duke is your one and only *reliable* task manager assistant!
Why should you download DukePro?
* Easy to Use
* Frequent Updates to suit YOUR needs
* ~~FAST~~ SUPERFAST to use!

How do you enlist Duke into your service?  **EASY!**
1.  Download <a href = "https://github.com/Droffilc13/ip/releases" target = "_blank">[here] </a>
2.  Double click
3.  Add your tasks
4.  Sit back and enjoy Duke manage your task for you!  :grinning:

## Features:
 - [x] Managing tasks
 - [ ] Managing deadlines (coming soon)
 - [ ] Reminders (coming soon)

## Custom Duke:
If you are a java programmer, feel free to customize Duke to suit your needs! 
Here is the `main` method:

```java
    public static void main(String[] args) {
        new Duke().run();
    }
```

Commit:
Add Gradle support

Droffilc13 committed:
Added basic functionality of chatbot and a Printer to format response of chatBot

Droffilc13 committed:
Added javaDocs for Printer and ChatBot class

Droffilc13 committed:
Added record and printList feature to ChatBot.java

Droffilc13 committed:
Fixed formatting of constant variable

Droffilc13 committed:
Added Task.java and mark as done functionality for tasks saved in Chatbot

Droffilc13 committed:
keywords bye and list are made case-sensitive

Droffilc13 committed:
Added Deadline, Event and Todo type of tasks

Droffilc13 committed:
Added Test Cases for ChatBot functionality

Droffilc13 committed:
Added functionality to prevent empty date in event and deadline task + relevant testcases

Droffilc13 committed:
Added deleteTask functionality for chatbot + added corresponding test cases

Droffilc13 committed:
Add save and retrieve tasks functionality

Droffilc13 committed:
Add date-time validation for deadline and event tasks

Droffilc13 committed:
Merge branch 'branch-Level-7'

Droffilc13 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/ChatBot.java
#	src/main/java/Deadline.java
#	src/main/java/Event.java

Droffilc13 committed:
Refactor Whole of Duke to Ui, Storage, TasksList, Parser Class

Droffilc13 committed:
Organise classes into different packages

Droffilc13 committed:
Delete outdated files

Droffilc13 committed:
Add JUnit test case for TaskList and Parser

Droffilc13 committed:
Create jar file for duke v0.1

Droffilc13 committed:
Add JavaDoc to most public methods and change interface name of  "Command" to "Commandable"

Droffilc13 committed:
Update JavaDocs to adhere to style requirements

Droffilc13 committed:
Tweak code to adhere to coding standards

Droffilc13 committed:
Add functionality to find all matching task that contains a particular word

Droffilc13 committed:
Merge branch 'branch-A-JavaDoc'

Droffilc13 committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/core/Duke.java
#	src/main/java/duke/core/Parser.java
#	src/main/java/duke/core/Storage.java
#	src/main/java/duke/core/TaskList.java
#	src/main/java/duke/core/Ui.java
#	src/main/java/duke/task/Task.java

Droffilc13 committed:
Merge branch 'branch-Level-9'

Droffilc13 committed:
Remove InvalidCommand class

Droffilc13 committed:
Add JavaDoc to changeStatusFromText method in Task class

Droffilc13 committed:
Change import statement in Parser class to meet coding standards

Droffilc13 committed:
Remove "Hello World" print statement in Duke

Droffilc13 committed:
Merge remote-tracking branch 'origin/add-gradle-support' into A-Gradle

Droffilc13 committed:
Add Gradle into project to build, run and test Duke

Droffilc13 committed:
Merge branch 'A-Gradle-Branch'

sebbycake commented on src/main/java/ChatBot.java:
```diff
@@ -6,9 +6,14 @@
 public class ChatBot {
     private boolean isRunning;
+    private String[] responses;
+    private static final int ResponseLimit = 100;
```
Perhaps the variable name for the constant can be changed to `RESPONSE_LIMIT` to adhere to Java coding standards!

sebbycake commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,38 @@
+ * @author Clifford
+ */
+public class Event extends Task {
+    protected final static String taskSymbol = "[E]";
```
Perhaps taskSymbol can be changed to `TASK_SYMBOL` as it is a constant.

Droffilc13 replied:
Thank you for taking the time to review my code. 
I have taken note of your comments and will make the necessary changes 😄

sebbycake reviewed (commented):
Hi Clifford. Perhaps you can add more javadocs for your Command subclasses and test classes. Otherwise, most of your code adheres to the Java coding standards. Good job!

Droffilc13 committed:
Convert Duke from text-based Ui to Gui

Droffilc13 committed:
Merge branch 'Level-10-Branch'

ruoyann commented on src/main/java/duke/command/Commandable.java:
```diff
@@ -0,0 +1,33 @@
+ *
+ * @author Clifford
+ */
+public interface Commandable {
```
Commandable doesn't sound very intuitive. Maybe command is better

Droffilc13 replied:
Thank you for taking the time to read through the code. I will consider changing to a more appropriate name. 😄

sharmainec commented on src/main/java/duke/core/Duke.java:
```diff
@@ -0,0 +1,81 @@
+     * @return the greeting message when user starts up Duke
+     */
+    public String greetUser() {
+        return "Hello! I'm Duke!\nHow may I be of service to you?";
```
I think that this method would fit well in Ui.java instead of having it here, would also be easier for you to change the text in the future since all the UI messages are grouped together :)

Droffilc13 replied:
That's a good point. Thank you for raising this.

sharmainec commented on src/main/java/duke/core/Parser.java:
```diff
@@ -0,0 +1,66 @@
+        String[] fullCommand = input.trim().split(" ", 2);
+        final String command = fullCommand[0].trim();
+        try {
+            if (command.equals("bye")) {
```
I like how your if blocks are very short and simple to read!

sharmainec commented on src/main/java/duke/task/Deadline.java:
```diff
@@ -0,0 +1,46 @@
+ * @author Clifford
+ */
+public class Deadline extends Task {
+    protected static final DateTimeFormatter PARSE_FORMAT = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
```
Great use of constants!! Helps to make the rest of your codes very clean! 👍

sharmainec commented on src/main/java/duke/task/Event.java:
```diff
@@ -0,0 +1,46 @@
+    protected static final DateTimeFormatter PARSE_FORMAT = DateTimeFormatter.ofPattern("d/M/yyyy HHmm");
+    protected static final DateTimeFormatter OUTPUT_FORMAT = DateTimeFormatter.ofPattern("dd MMM yyyy, hh:mm a");
+    protected LocalDateTime dateTime;
+
```
Is this an extra line? (I'm not too sure...)

Droffilc13 replied:
Indeed it is. I have made the necessary corrections

sharmainec reviewed (commented):
LGTM. The codes were very simple to follow! 👍

Droffilc13 committed:
Change minor syntax with AddEventCommand and Event and gradle

Droffilc13 committed:
Change Commandable to Comand

Droffilc13 committed:
Add assertions in Duke and Storage class

Droffilc13 committed:
Delete Dead code in Duke class

run() and main() method in Duke are not used.

run() and main() method are not needed for the current Gui-based Duke
when we transitioned from text-based Duke.

Let's delete the run() and main() method in Duke.

Deleting dead code makes the code base cleaner and easier to maintain.

Droffilc13 committed:
Refactor TaskList, Ui and Task class

Droffilc13 committed:
Change Parser to include a default statement for switch statement

Droffilc13 committed:
Extract method of saving file in saveFile method of Storage class

saveFile method has a method implementation to write to file.

The method implementation may have violate SLAP..

Let's extract out the method implementation and place it in a separate
method called writeToFile.

This will make it easier to read and debug in the future.

Droffilc13 committed:
Add Javadoc for greetUser and showError method

Droffilc13 committed:
Merge pull request #1 from Droffilc13/branch-A-Assertions

Add Assertions in to Duke and Storage class

Droffilc13 committed:
Merge branch 'master' into branch-A-CodeQuality

Droffilc13 committed:
Merge pull request #2 from Droffilc13/branch-A-CodeQuality

Improve code quality of classes within Duke application

Droffilc13 committed:
Add gradle.yml file to set up Conitnuous Integration

Droffilc13 committed:
Remove try-catch block in Parser

IdentifyCommand method in Parser has a try-catch block for cases when a user
does not specify additional required arguments.

There is an inconsistency in some Commands returning their own custom
message if user does not enter additional arguments before a switch
statement was used.

Let's standardize it across the commands for all of them to return
custom messages if user is short of arguments required for command.

It is easier to find out what user argument is lacking from a specified
message rather than a catch-all message.

Droffilc13 committed:
Update ParserTest test cases

Droffilc13 committed:
Add sort functionality to task list

Task list currently displays tasks as they were added in.

A sorted chronological list may help users see which tasks are upcoming.

Let's,
* add a sort functionality to TaskList
* add a test case in TaskListText

sortTask uses Comparator to sort the Tasks and places earlier task at the top
since users are more likely to look at earlier tasks. todo task is assumed to have
less urgency since they do not have a date attached to it.

Droffilc13 committed:
Remove unused import statements in Deadline and Event class

Droffilc13 committed:
Transfer formatting dates from deadline and event class to parent Task class

Droffilc13 committed:
Merge pull request #3 from Droffilc13/branch-C-Sort

Add sort functionality to task list

Droffilc13 committed:
Refactor code in Task List for clarity

Droffilc13 committed:
Improve Duke Gui design

Droffilc13 committed:
Personalize Duke to by changing messages print to user

Droffilc13 committed:
Change title of application

Droffilc13 committed:
Change stage title of application

Droffilc13 committed:
Change title of stage of application to "Dragon Task Maid | Tohru"

Droffilc13 committed:
Change title of application to "Dragon Maid | Tohru"

Droffilc13 committed:
Change list command dialogue

Droffilc13 committed:
Set up representative screenshot of duke

Droffilc13 committed:
Update representative screenshot of Duke

Droffilc13 committed:
Update README.md with latest features and usage

Droffilc13 committed:
Change stage title of the application back to 'Duke'

Droffilc13 committed:
Change boolean variable name to fit coding standards

Droffilc13 committed:
Fix image naming

Droffilc13 committed:
Change title of application

Droffilc13 committed:
Crop Ui.png to remove black line at left of image

Droffilc13 committed:
Change Ui.png due to excessive cropping at application edges

Droffilc13 committed:
Update README.md to change title
